clients/ts/FunctionalTests/ts/HubConnectionTests.ts

```diff
@@ -379,9 +379,7 @@ describe("hubConnection", () => {
                     ByteArray: protocol.name === "json"
                         ? "aGVsbG8="
                         : new Uint8Array([0x68, 0x65, 0x6c, 0x6c, 0x6f]),
-                    GUID: protocol.name === "json"
-                        ? "00010203-0405-0607-0706-050403020100"
-                        : new Uint8Array([0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x07, 0x06, 0x05, 0x04, 0x03, 0x02, 0x01, 0x00]),
```

Comment:
Dates are a SOB when serializing, especially bad in .NET because `DateTime` pretends to know about time zones.

I think it would be useful to have a range of tests for serializing and deserializing DateTime with different DateTime.Kind values, and DateTimeOffset.

It's complex and fiddly and a pain in the butt, but unfortunatly important.

Reply:
(and if you mess it up in the beginning you can't easily change how time zones are handled without breaking everyone in subtle and hard to diagnose ways)

Reply:
>DateTime is serialized to MessagePack Timestamp format, it serialize/deserialize UTC and loses Kind info.

I used the native datetime formatter to solve this issue, but that brings other issues. I'm not sure how we can do datetime then...

Reply:
Oh yeah, I do remember that. We could serialize to strings I suppose